src/eviction/eviction.c

```diff
@@ -42,60 +42,66 @@ static uint32_t ocf_evict_calculate(struct ocf_user_part *part,
 
 static inline uint32_t ocf_evict_do(ocf_cache_t cache,
 		ocf_queue_t io_queue, const uint32_t evict_cline_no,
-		ocf_part_id_t target_part_id)
+		ocf_part_id_t target_part_id, bool part_evict)
 {
 	uint32_t to_evict = 0, evicted = 0;
 	struct ocf_user_part *part;
 	struct ocf_user_part *target_part = &cache->user_parts[target_part_id];
 	ocf_part_id_t part_id;
 
-	/* For each partition from the lowest priority to highest one */
-	for_each_part(cache, part, part_id) {
+	if (part_evict) {
+		evicted = ocf_eviction_need_space(cache, io_queue,
+				target_part_id, evict_cline_no);
```

Comment:
Returning value here will let You avoid using `else` and making another indentation block